src/main/java/com/mindera/school/spaceshiprent/service/user/UserServiceImpl.java

```diff
@@ -14,16 +15,27 @@
 import java.util.Optional;
 import java.util.stream.Collectors;
 
+import static com.mindera.school.spaceshiprent.exception.ErrorMessageConstants.USER_ALREADY_EXISTS;
+import static com.mindera.school.spaceshiprent.exception.ErrorMessageConstants.USER_NOT_FOUND;
+
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
 
     private final UserConverter converter;
     private final UserRepository userRepository;
+    private final EmailSender emailSender;
 
     @Override
     public UserDetailsDto createUser(CreateOrUpdateUserDto createOrUpdateUserDto) {
+        userRepository.findByEmail(createOrUpdateUserDto.getEmail())
+                .ifPresent(user -> {
+                    throw new UserAlreadyExists(USER_ALREADY_EXISTS);
+                });
+
         UserEntity userEntity = converter.convertToEntity(createOrUpdateUserDto);
+        String emailingInfo = userEntity.getEmail() + " " + userEntity.getName();
```

Comment:
Don't like this String here... you're basically saying that UserService should know the format that Mailjet needs in order to send emails. That should be Mailjet responsibility.
I suggest your method `emailSender.send(...)` to receive the name and email as params then inside that method it builds the string, or wherever makes more sense in terms of logic responsibility.

Reply:
you're right, thank you